src/System.IO.Pipelines.Networking.Sockets/SocketConnection.cs

```diff
@@ -598,13 +599,14 @@ private void RecycleSmallBuffer(ref ArraySegment<byte> buffer)
         }
 
         // unsafe+async not good friends
-        private unsafe void SetBuffer(Memory<byte> memory, SocketAsyncEventArgs args, int ignore = 0)
+        private void SetBuffer(ReadOnlyMemory<byte> readOnlyMemory, SocketAsyncEventArgs args, int ignore = 0)
         {
-            if (!memory.TryGetArray(out ArraySegment<byte> segment))
+            var memory = Unsafe.As<ReadOnlyMemory<byte>, Memory<byte>>(ref readOnlyMemory);
```

Comment:
Take a look at https://github.com/dotnet/corefx/blob/master/src/System.Memory/ref/System.Memory.cs#L437

Reply:
I don’t like this change.